An assertion has failed: Expected indention for a prefixed line

```
workspace/pulumi/hackathon2                                                                                                                                                                                  
▶ pulumi update -p 20
Previewing update of stack 'mastodon-dev'
Previewing changes:

     Type                 Name                   Plan          Info
 *   pulumi:pulumi:Stack  mastodon-mastodon-dev  no change

info: no changes required:
      55 resources unchanged

Do you want to perform this update? yes
Updating stack 'mastodon-dev'
Performing changes:

     Type                 Name                   Status     Info
 *   pulumi:pulumi:Stack  mastodon-mastodon-dev  done
panic: fatal: An assertion has failed: Expected indention for a prefixed line

goroutine 544 [running]:
github.com/pulumi/pulumi/pkg/util/contract.failfast(0xc42136c040, 0x3f)
        /Users/swgillespie/go/src/github.com/pulumi/pulumi/pkg/util/contract/failfast.go:23 +0xe9
github.com/pulumi/pulumi/pkg/util/contract.Assertf(0xc4202f4200, 0x1a9b517, 0x26, 0x0, 0x0, 0x0)
        /Users/swgillespie/go/src/github.com/pulumi/pulumi/pkg/util/contract/assert.go:33 +0x126
github.com/pulumi/pulumi/pkg/engine.getIndentationString(0x0, 0x1a799a3, 0x6, 0x1, 0x3, 0xc4219217e8)
        /Users/swgillespie/go/src/github.com/pulumi/pulumi/pkg/engine/diff.go:72 +0x198
github.com/pulumi/pulumi/pkg/engine.writeWithIndent(0xc4202f42a0, 0x0, 0x1a799a3, 0x6, 0x1, 0xc4219217e8, 0x7, 0xc421921820, 0x1, 0x1)
        /Users/swgillespie/go/src/github.com/pulumi/pulumi/pkg/engine/diff.go:94 +0x86
github.com/pulumi/pulumi/pkg/engine.printPropertyTitle(0xc4202f42a0, 0xc421457510, 0xa, 0xa, 0x0, 0x1a799a3, 0x6, 0x1)
        /Users/swgillespie/go/src/github.com/pulumi/pulumi/pkg/engine/diff.go:319 +0x179
github.com/pulumi/pulumi/pkg/engine.printObjectPropertyDiff.func1(0x1a799a3, 0x6, 0x1)
        /Users/swgillespie/go/src/github.com/pulumi/pulumi/pkg/engine/diff.go:451 +0x6b
github.com/pulumi/pulumi/pkg/engine.printAdd(0xc4202f42a0, 0x1a09400, 0xc42100e8a0, 0xc421921968, 0xc4209e3100, 0x0, 0x0)
        /Users/swgillespie/go/src/github.com/pulumi/pulumi/pkg/engine/diff.go:583 +0x44
github.com/pulumi/pulumi/pkg/engine.printObjectPropertyDiff(0xc4202f42a0, 0xc421457510, 0xa, 0xa, 0xc421377e60, 0xc421377e90, 0xc421377ec0, 0xc421377ef0, 0x0, 0x0, ...)
        /Users/swgillespie/go/src/github.com/pulumi/pulumi/pkg/engine/diff.go:454 +0x132
github.com/pulumi/pulumi/pkg/engine.GetResourceOutputsPropertiesString(0x1a77ea3, 0x4, 0xc42137d180, 0x4d, 0xc42124fac0, 0x13, 0xc42178cee0, 0xc42178d030, 0xc42178d1f0, 0x0, ...)
        /Users/swgillespie/go/src/github.com/pulumi/pulumi/pkg/engine/diff.go:275 +0x2bd
github.com/pulumi/pulumi/pkg/backend/local.(*ProgressDisplay).processEndSteps(0xc42181df00)
        /Users/swgillespie/go/src/github.com/pulumi/pulumi/pkg/backend/local/progress.go:709 +0x962
github.com/pulumi/pulumi/pkg/backend/local.(*ProgressDisplay).processEvents(0xc42181df00, 0xc4209da780, 0xc4212994a0)
        /Users/swgillespie/go/src/github.com/pulumi/pulumi/pkg/backend/local/progress.go:961 +0x173
github.com/pulumi/pulumi/pkg/backend/local.DisplayProgressEvents.func1(0xc42181df00, 0xc4209da780, 0xc4212994a0, 0xc421299500)
        /Users/swgillespie/go/src/github.com/pulumi/pulumi/pkg/backend/local/progress.go:264 +0x3f
created by github.com/pulumi/pulumi/pkg/backend/local.DisplayProgressEvents
        /Users/swgillespie/go/src/github.com/pulumi/pulumi/pkg/backend/local/progress.go:263 +0x3ee
```

I hit this as well doing a `pulumi up`.  In my case it was:

```
Diagnostics:
  kubernetes:core:Service: frontend
    warning: This resource failed to initialize in a previous deployment. It is recommended
             to update it to fix these issues:
               - Problem #1 Timeout occurred for 'frontend'
               - Problem #2 Service does not target any Pods
 
panic: fatal: An assertion has failed: Expected indention for a prefixed line

goroutine 267 [running]:
github.com/pulumi/pulumi/pkg/util/contract.failfast(0xc4204e5240, 0x3f)
	/home/cyrusn/go/src/github.com/pulumi/pulumi/pkg/util/contract/failfast.go:23 +0xf3
github.com/pulumi/pulumi/pkg/util/contract.Assertf(0xc420224000, 0xf01cb6, 0x26, 0x0, 0x0, 0x0)
	/home/cyrusn/go/src/github.com/pulumi/pulumi/pkg/util/contract/assert.go:33 +0x13d
github.com/pulumi/pulumi/pkg/engine.getIndentationString(0x0, 0xee0643, 0x6, 0x1, 0x3, 0xc420feb7e0)
	/home/cyrusn/go/src/github.com/pulumi/pulumi/pkg/engine/diff.go:72 +0x1a6
github.com/pulumi/pulumi/pkg/engine.writeWithIndent(0xc420224000, 0x0, 0xee0643, 0x6, 0x1, 0xc420feb7e0, 0x7, 0xc420feb818, 0x1, 0x1)
	/home/cyrusn/go/src/github.com/pulumi/pulumi/pkg/engine/diff.go:94 +0x86
```

--- a/pkg/backend/local/progress.go
+++ b/pkg/backend/local/progress.go
@@ -702,19 +702,17 @@ func (display *ProgressDisplay) processEndSteps() {
 		}
 	}
 
-	// If we're not previewing, and we get stack outputs, then display them at the end.
-	if !display.isPreview {
-		if display.stackUrn != "" {
-			stackStep := display.eventUrnToResourceRow[display.stackUrn].Step()
-			props := engine.GetResourceOutputsPropertiesString(stackStep, 0, false, display.opts.Debug)
-			if props != "" {
-				if !wroteDiagnosticHeader {
-					display.writeBlankLine()
-				}
-
-				wroteDiagnosticHeader = true
-				display.writeSimpleMessage(props)
+	// If we get stack outputs, display them at the end.
+	if display.stackUrn != "" {
+		stackStep := display.eventUrnToResourceRow[display.stackUrn].Step()
+		props := engine.GetResourceOutputsPropertiesString(stackStep, 1, display.isPreview, display.opts.Debug)
+		if props != "" {
+			if !wroteDiagnosticHeader {
+				display.writeBlankLine()
 			}
+
+			wroteDiagnosticHeader = true
+			display.writeSimpleMessage(props)
 		}
 	}
 

--- a/pkg/engine/diff.go
+++ b/pkg/engine/diff.go
@@ -252,10 +252,16 @@ func GetResourceOutputsPropertiesString(
 		outputDiff = step.Old.Outputs.Diff(outs)
 	}
 
+	var keys []resource.PropertyKey
+	if outputDiff == nil {
+		keys = outs.StableKeys()
+	} else {
+		keys = outputDiff.Keys()
+	}
+	maxkey := maxKey(keys)
+
 	// Now sort the keys and enumerate each output property in a deterministic order.
 	firstout := true
-	keys := outs.StableKeys()
-	maxkey := maxKey(keys)
 	for _, k := range keys {
 		out := outs[k]
 

--- a/tests/integration/integration_test.go
+++ b/tests/integration/integration_test.go
@@ -187,8 +187,8 @@ func TestStackOutputsDisplayed(t *testing.T) {
 			output := stdout.String()
 
 			// ensure we get the --outputs:-- info both for the normal update, and for the no-change update.
-			assert.Contains(t, output, "---outputs:---\nfoo: 42\nxyz: \"ABC\"\n\ninfo: 1 change performed")
-			assert.Contains(t, output, "---outputs:---\nfoo: 42\nxyz: \"ABC\"\n\ninfo: no changes required")
+			assert.Contains(t, output, "---outputs:---\n    foo: 42\n    xyz: \"ABC\"\n\ninfo: 1 change performed")
+			assert.Contains(t, output, "---outputs:---\n    foo: 42\n    xyz: \"ABC\"\n\ninfo: no changes required")
 		},
 	})
 }

